### ISSUE TITLE
Getting an error after updating to build 0.3.2 

### ISSUE THREAD

**Issue by igorstajic** [Vestorly/torii#156]
The error is happening inside ember-addon\index.js
Its unable to get **app** reference from `this.app.env` (line 26).
And also from `this.app.name`(line 37).

I've managed to work around this by hard-coding the project name instead of `this.app.name`,
and replacing `this.app.env` with `this.project.env`.

I'm quite new to the whole ember-cli thing, so maybe I was doing something wrong when installing this package. Any help would be appreciated :)

Here is the whole error stack:
```
Cannot read property 'env' of undefined
TypeError: Cannot read property 'env' of undefined
   `at Class.treeForAddon (D:\Ember_practice\my-first-app\node_modules\torii\ember-addon\index.js:26:46)
    at Class._treeFor (D:\Ember_practice\my-first-app\node_modules\ember-cli\lib\models\addon.js:288:31)
    at Class.treeFor (D:\Ember_practice\my-first-app\node_modules\ember-cli\lib\models\addon.js:258:19)
    at D:\Ember_practice\my-first-app\node_modules\ember-cli\lib\models\addon.js:211:32
    at Array.map (native)
    at Class.eachAddonInvoke (D:\Ember_practice\my-first-app\node_modules\ember-cli\lib\models\addon.js:209:22)
    at Class.treeFor (D:\Ember_practice\my-first-app\node_modules\ember-cli\lib\models\addon.js:256:20)
    at EmberApp.<anonymous> (D:\Ember_practice\my-first-app\node_modules\ember-cli\lib\broccoli\ember-app.js:307:20)
    at Array.map (native)
    at EmberApp.addonTreesFor (D:\Ember_practice\my-first-app\node_modules\ember-cli\lib\broccoli\ember-app.js:305:30)
```

**Comment by Frozenfire92** [comment]
Also getting this error

**Comment by mixonic** [comment]
What version of node and ember are you using? `ember --version` should output this info.

The output of `npm ls|grep -i torii` would also be appreciated.

**Comment by Frozenfire92** [comment]
I was using
```
node v0.10.33
ember version: 0.2.0
```
However in my package.json I am using `"ember-cli": "ember-cli/ember-cli",` to fix stupid errors introduced with 0.2.0


```
├─┬ ember-cli-simple-auth-torii@0.7.3
│ └── torii@0.1.3
└─┬ torii@0.3.2
```

I have just updated node and npm and the issue persists

**Comment by mixonic** [comment]
@Frozenfire92 bug are bugs, we don't need to insult their intelligence :-p

It looks like ember-cli-simple-auth-torii is still using Torii 0.1.3. You will need to update this dependency to use Torii with Ember-CLI 0.2.0.

**Comment by Frozenfire92** [comment]
> bug are bugs, we don't need to insult their intelligence :-p

I had a good chuckle. Thanks :)

**Comment by igorstajic** [comment]
I am still getting the error. Even tho I updated the dependency. The `ls npm | grep -i torii` output now is 
`├── ember-cli-simple-auth-torii@0.7.3`
`└─┬ torii@0.3.2`

**Comment by timini** [comment]
yep me too

**Comment by mixonic** [comment]
can someone push a reproduction? This is the same error pre-0.3.2 versions of Torii would show with Ember-CLI 0.2.0, so I remain suspicious of npm issues. Please try `rm -rf node_modules/ && npm install`.

**Comment by igorstajic** [comment]
I've tried removing the `node_modules` folder, but still getting the same error. `npm install` went through fine.

**Comment by he9lin** [comment]
tried removing `node_modules` and do `npm install` again. still getting this error :(  I browse through a bit of ember-cli source code `/lib/models/addon.js`, no idea why `this.app` is undefined in `torii/ember-addon/index.js`

```
├─┬ ember-cli-simple-auth-torii@0.7.3 (git+https://github.com/simplabs/ember-cli-simple-auth-torii#37eac35a8a7aa95945ead0b89c7268c31f14644c)
│ └─┬ torii@0.3.2
```

my `package.json`
```json
{
  "name": "megafans",
  "version": "0.0.0",
  "description": "Small description for megafans goes here",
  "private": true,
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "scripts": {
    "start": "ember server",
    "build": "ember build",
    "test": "ember test"
  },
  "repository": "",
  "engines": {
    "node": ">= 0.10.0"
  },
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "broccoli-asset-rev": "^2.0.0",
    "broccoli-sass": "0.6.2",
    "ember-cli": "ember-cli/ember-cli",
    "ember-cli-app-version": "0.3.2",
    "ember-cli-babel": "^4.0.0",
    "ember-cli-coffeescript": "0.10.0",
    "ember-cli-content-security-policy": "0.3.0",
    "ember-cli-dependency-checker": "0.0.8",
    "ember-cli-font-awesome": "0.0.9",
    "ember-cli-htmlbars": "0.7.4",
    "ember-cli-ic-ajax": "0.1.1",
    "ember-cli-inject-live-reload": "^1.3.0",
    "ember-cli-qunit": "0.3.9",
    "ember-cli-simple-auth": "^0.7.3",
    "ember-cli-simple-auth-devise": "0.7.3",
    "ember-cli-simple-auth-torii": "simplabs/ember-cli-simple-auth-torii",
    "ember-cli-uglify": "1.0.1",
    "ember-data": "1.0.0-beta.15",
    "ember-export-application-global": "^1.0.2"
  },
  "dependencies": {
    "nprogress": "^0.1.6"
  }
}
```

**Comment by soundarapandian** [comment]
Same here. I get this issue even though I use Torii 0.3.2. How do I get out of this issue?

```
npm ls|grep -i torii
├─┬ ember-cli-simple-auth-torii@0.7.3 (git+https://github.com/simplabs/ember-cli-simple-auth-torii#37eac35a8a7aa95945ead0b89c7268c31f14644c)
│ └─┬ torii@0.3.2
```

**Comment by timini** [comment]
anyone know a fix?

**Comment by ashrafhasson** [comment]
I'm having the same with ember-cli#0.2.1, I think this issue needs reopening?

**Comment by lypse** [comment]
I'm also having this issue with ember-cli 0.2.1 and Torii 0.3.2

**Comment by mixonic** [comment]
Is everyone using Ember-Simple-Auth?

I just created a new Ember-CLI 0.2.1 project, then installed Torii 0.2.1. It builds correctly. Without a reproduction or more information, this will not be resolved. Can anyone create a reproduction?

**Comment by igorstajic** [comment]
The error starts to happen after installing *ember-cli-simple-auth-torii* package. 
On a new project, after installing Torii 0.3.2 and ember-cli-simple-auth, it builds fine. But after that installing *ember-cli-simple-auth-torii* leads to the error.

**Comment by ashrafhasson** [comment]
Created a new ember-cli#0.2.1 project > installed ember-cli-simple-auth#0.7.3 > installed torii#0.3.2 > build's fine. Installed ember-cli-simple-auth-torii#0.7.3 (this ships with torii#0.1.3) > build failed with `cannot read property 'env' of undefined` > updated ember-cli-simple-auth-torii's torii dependency to 0.3.2 > build still fails with the same error message.

**Comment by mixonic** [comment]
ember-cli-simple-auth 0.7.3 is dependent upon Torii 0.1.3. See:

* https://github.com/simplabs/ember-cli-simple-auth-torii/blob/0.7.3/package.json#L20

So I presume you all have an old version of Torii installed. **Torii 0.1.3 is not compatible with Ember-CLI 0.2.x**. npm is weird, and it will install the 0.1.3 version and use it even if your package.json has Torii 0.3.2 listed as a direct dependency.

Please report issues with that package at the ember-cli-simple-auth repo. I believe @marcoow is waiting on something before making a release.

**Comment by marcoow** [comment]
I plan to release Ember Simple Auth 0.8.0 (or a beta of it) with the updated torii dependency until next week.

**Comment by ashrafhasson** [comment]
:+1: thanks @mixonic and @marcoow

**Comment by timini** [comment]
@mixonic We are all using the master of simplabs/ember-cli-simple-auth-torii with the updated dependancy and we are still getting this error.

**Comment by dbachet** [comment]
@mixonic @marcoow I confirm. Here are the three linked issues that are closed but seem to pose problem to a bunch of people. 

https://github.com/simplabs/ember-cli-simple-auth-torii/issues/10

https://github.com/Vestorly/torii/issues/147

https://github.com/Vestorly/torii/issues/156 (this is the current issue)

The only found workaround is to rollback to simple-auth 0.6.7 and torii 0.2.1

(I don't know how else I could help, any idea to debug that deeper so that I can try to do something more?)

**Comment by Nestor-at-ps** [comment]
Finally @mixonic fixed the bug, in 0.3.3 https://github.com/Vestorly/torii/commit/4c3dff20b54644facaedade6c0c92ed253b6e22b

-    var config = this.project.config(proccess.env.EMBER_ENV);
+    var config = this.project.config(process.env.EMBER_ENV);

I kindly suggest that next time devs take more time investigating a bug instead of blaming  other packages, this bug could be fixed 20 days ago.

**Comment by mixonic** [comment]
@Nestor-at-ps that line is not present in the 0.3.2 release. The typo was only in master between 0.3.2 and 0.3.3, and was fixed by the time 0.3.3 was released. I doubt it was related, or if it was then it seems people were installing master and not 0.3.2.

**Comment by Nestor-at-ps** [comment]
@mixonic  Thanks for the clarification. You are right the typo was not released, still I was using 0.3.2 (not master) and got the error. So, for another reason, this.app.env (line 26) wasn't recognized.  

This gave me the hint.
https://github.com/simplabs/ember-cli-simple-auth-torii/issues/13  

Thanks for the new version I'll try it later, Torri has helped me a lot.
